client/components/data/query-site-monitor-settings/index.jsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingSiteMonitorSettings } from 'state/selectors';
+import { requestSiteMonitorSettings } from 'state/sites/monitor/actions';
+
+class QuerySiteMonitorSettings extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		requestingSiteMonitorSettings: PropTypes.bool,
+		requestSiteMonitorSettings: PropTypes.func
+	};
+
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		if ( props.requestingSiteMonitorSettings ) {
```

Comment:
We can maybe save a network request if we check if the site is a jetpack site, but optional since the request just returns an error in that case.